yb-voyager/cmd/endMigrationCommand.go

```diff
@@ -178,47 +189,99 @@ func saveDataMigrationReport(msr *metadb.MigrationStatusRecord) {
 		fmt.Sprintf("FF_DB_PASSWORD=%s", fallForwardDBPassword),
 		fmt.Sprintf("SOURCE_DB_PASSWORD=%s", sourceDBPassword),
 	}
-	liveMigrationReportFilePath := filepath.Join(backupDir, "reports", "data_migration_report.txt")
-	strCmd := fmt.Sprintf("yb-voyager get data-migration-report --export-dir %s > %q", exportDir, liveMigrationReportFilePath)
+
+	getDataMigrationReportPath := filepath.Join(backupDir, "reports", "data_migration_report.txt")
+	strCmd := fmt.Sprintf("yb-voyager get data-migration-report --export-dir %s", exportDir)
 	liveMigrationReportCmd := exec.Command("bash", "-c", strCmd)
 	liveMigrationReportCmd.Env = append(os.Environ(), passwordsEnvVars...)
-	var outbuf bytes.Buffer
+	var outbuf, errbuf bytes.Buffer
 	liveMigrationReportCmd.Stderr = &outbuf
+	liveMigrationReportCmd.Stdout = &errbuf
+
 	err := liveMigrationReportCmd.Run()
 	if err != nil {
-		log.Errorf("running get data-migration-report command: %s: %v", outbuf.String(), err)
+		log.Errorf("running get data-migration-report command: %s: %v", errbuf.String(), err)
 		utils.ErrExit("running get data-migration-report command: %v", err)
 	}
+
+	outbufBytes := bytes.Trim(outbuf.Bytes(), " \n")
+	outbufBytes = append(outbufBytes, []byte("\n")...)
+	// overwrite existing if there is a new report
+	if len(outbufBytes) > 0 {
+		err = os.WriteFile(getDataMigrationReportPath, outbufBytes, 0644)
+		if err != nil {
+			utils.ErrExit("writing data migration report: %v", err)
+		}
+	} else {
+		if utils.FileOrFolderExists(getDataMigrationReportPath) {
+			utils.PrintAndLog("data migration report is already present at %q", getDataMigrationReportPath)
+		}
+	}
 }
 
-func saveDataExportImportReports(msr *metadb.MigrationStatusRecord) {
+func saveDataExportReport() {
```

Comment:
Same comments as given to the `saveDataMigrationReport()` apply here as well.

How about factoring out the common code structure as `saveCommandOutput(cmd, filePath)`?